### PR TITLE
Fixes #22888 - remove code around dynflow db_pool_size

### PR DESCRIPTION
**Pull request by iNecas**
Remove the code in favor of the Dynfow, where we can handle all the
logic at one place.

Requires https://github.com/Dynflow/dynflow/pull/277

### CODE REVIEW

**Comment by iNecas** [comment]
Setting to WOC until the new Dynflow is released

**Review by adamruzicka — CHANGES_REQUESTED** [review]
Works as expected, pending dynflow dependency version bump

**Comment by iNecas** [comment]
Version dependency dumped, waiting for dynflow to be released

**Comment by iNecas** [comment]
New dynflow released: ready for final review

**Comment by iNecas** [comment]
@ares searching for core committer for merging this one.

**Comment by ares** [comment]
[test foreman] [test katello] [test upgrade]

File: Gemfile [hunk 1]
```diff
@@ -50,7 +50,7 @@ gem 'deacon', '~> 1.0'
 gem 'webpack-rails', '~> 0.9.8'
 gem 'mail', '~> 2.7'
 gem 'sshkey', '~> 1.9'
-gem 'dynflow', '>= 0.8.29', '< 1.0.0'
+gem 'dynflow', '>= l.0.0', '< 2.0.0'
```
Comment (mmoll) [inline Gemfile]: 1 instead of l, I presume? ;)

Comment (iNecas) [reply Gemfile]: LOL, close enough :) Sry, I guess I need to take some rest, updated

**Review by mmoll — APPROVED** [review]

**Comment by mmoll** [comment]
merged, díky @iNecas!

**Pull request merged**